microprofile-fault-tolerance/src/test/java/org/jboss/eap/qe/microprofile/fault/tolerance/UndeployDeployTest.java

```diff
@@ -0,0 +1,307 @@
+import org.jboss.shrinkwrap.api.asset.StringAsset;
+import org.jboss.shrinkwrap.api.spec.WebArchive;
+import org.junit.After;
+import org.junit.AfterClass;
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.wildfly.extras.creaper.core.online.OnlineManagementClient;
+
+/**
+ * Tests deploy/undeploy of MP FT deployments (WAR archives)
+ */
+@RunAsClient
+@RunWith(Arquillian.class)
+public class UndeployDeployTest {
+
+    private static final String FIRST_DEPLOYMENT = "UndeployDeployTest-first-deployment";
+    private static final String SECOND_DEPLOYMENT = "UndeployDeployTest-second-deployment";
+    private static final String NO_MP_FT_DEPLOYMENT = "no-mp-ft-deployment";
+
+    @ArquillianResource
+    private Deployer deployer;
+
+    @Deployment(name = FIRST_DEPLOYMENT, managed = false)
+    public static Archive<?> createFirstDeployment() {
+        String mpConfig = "hystrix.command.default.execution.timeout.enabled=true";
+
+        return ShrinkWrap.create(WebArchive.class, FIRST_DEPLOYMENT + ".war")
+                .addPackages(true, HelloService.class.getPackage())
+                .addAsManifestResource(EmptyAsset.INSTANCE, "beans.xml")
+                .addAsManifestResource(new StringAsset(mpConfig), "microprofile-config.properties");
+    }
+
+    @Deployment(name = SECOND_DEPLOYMENT, managed = false)
+    public static Archive<?> createSecondDeployment() {
+        String mpConfig = "hystrix.command.default.execution.timeout.enabled=false";
+
+        return ShrinkWrap.create(WebArchive.class, SECOND_DEPLOYMENT + ".war")
+                .addPackages(true, HelloService.class.getPackage())
+                .addAsManifestResource(EmptyAsset.INSTANCE, "beans.xml")
+                .addAsManifestResource(new StringAsset(mpConfig), "microprofile-config.properties");
+    }
+
+    @Deployment(name = NO_MP_FT_DEPLOYMENT, managed = false, testable = false)
+    public static Archive<?> createNonMPFTDeployment() {
+        return ShrinkWrap.create(WebArchive.class, NO_MP_FT_DEPLOYMENT + ".war")
+                .addPackage(org.jboss.eap.qe.microprofile.fault.tolerance.deployments.nofaulttolerance.HelloService.class
+                        .getPackage())
+                .addAsManifestResource(EmptyAsset.INSTANCE, "beans.xml");
+    }
+
+    @BeforeClass
+    public static void setup() throws Exception {
+        MicroProfileFaultToleranceServerConfiguration.enableFaultTolerance();
+    }
+
+    /**
+     * Deploy all deployments so it's possible to get their URLs in other tests. This is limitation of Arquillian
+     * when deployments are deployed manually (in tests) and thus their URL is not known in time when test is started.
+     * The only way how to workaround this issue is to deploy all deployments in first test which allows to inject
+     * deployment URL as params in other test methods.
+     */
+    @Test
+    @InSequence(1)
```

Comment:
I understand that this "test method" must execute first. Is the rest of tests dependent on each other somehow? If not, it would be IMHO better to have same "priority" for rest of tests and leave this at lower one. WDYT?

Reply:
+1, fixed